### ISSUE TITLE
Downloads of Drush > 7 are broken 

### ISSUE THREAD

**Issue by sint** [drush-ops/drush#1719]
When downloading a tar.gz from drush release page you'll end up with the following error:

**Unable to load autoload.php. Drush now requires Composer in order to install its dependencies and autoload classes. Please see README.md**

All of the necessary "include" and "require" statements are missing to run Drush when downloading those Tar files. Please inlcude autoloading independent from composer.

Since Drush is software you'll most likely want to install on a server it would be a good way to have it independed from composer. This is what prevents package maintainers of Linux to update Drush within their repositories. Gentoo Linux had fine support of Drush up to 6.6. and dropped support for Drush 7 and 8 because of this!

**Comment by kenorb** [comment]
Related: #1718

**Comment by weitzman** [comment]
Like Drupal and the rest of the PHP ecosystem, we depend on Composer now. You are shooting the messenger.

**Comment by sint** [comment]
Its just sad to see Drush going into a one way road. I don't see any problem with Drupal forcing people to use inferior package managers. You'll end up with one Drupal on one project anyways. Why should people install one Drush for each project? It was your decision to drop the possibility of clean inclusion into existing package management systems, including those having perfect Drush support for years. Just provide static builds and things will keep going nicely.

**Comment by sint** [comment]
I know you can install it on System Level but with some 3rd party package manager and loosing all the good parts linux has provided for years.

**Comment by kenorb** [comment]
@sint You can always install it via Apt: `sudo apt-get install drush`, it won't require Composer, but you'll get slightly [older version](https://github.com/drush-ops/drush/issues/1628).

**Comment by sint** [comment]
@kenorb: Thanks allot for your efforts to convince me that eveything is ok. I really appreciate it! 

I am not using Debian / Ubuntu and for that reason i can not use Apt. I am using Gentoo with Portage. They have a highly motivated maintainer supporting Drush up until 6.6 but have decided to drop support of Drush for security reasons.

I've had some chat with the maintainer about combining Composer with Portage. He explained the situation in Detail:

----

The problem is that composer bundles all of the dependencies for a given package along with that package. That can easily blow up to hundreds of packages "copy/pasted" into yours, since it also fetches dependencies of dependencies, and dependencies of those dependencies of dependencies...

That's not strictly forbidden by Gentoo, but it's strongly discouraged. It's also a big weakness in terms of security, especially for PHP projects that often sit on the public internet. If there's a security hole in one of Drush's dependencies, you'd never know it. (Drush isn't public-facing, but you get the idea.)

To make everything work, you would have to install Drush locally using Composer, and figure out exactly what it needs for its dependencies. Then you'd have to modify the Composer autoload magic script to look in the normal system location for those dependencies, and remove the ones that Composer has bundled. Finally, you'd have to create a tarball snapshot of the installation on your local machine and use that as the SRC_URI for the ebuild.

And you'd have to do that for every new version. It's too much trouble if it's even possible... the composer-installed copy may not like being moved around, and that would cause even more headaches. I tried to do it a few months ago and gave up.

---

This situation would most likely be similar to any Linux Distribution.

Good news is, that after i got bluntly rejected here, i've found out about efforts of Drush Phar builds. That way, it is possible again to install Drush without composer by "just download" a package. I've talked back to the maintainer and he assured me, he will look into it when they get released.
